llvm/lib/Target/STM8/STM8ISelLowering.cpp

```diff
@@ -23,6 +23,26 @@ namespace llvm {
 STM8TargetLowering::STM8TargetLowering(const STM8TargetMachine &TM,
                                        const STM8Subtarget &STI)
     : TargetLowering(TM) {
+
+  // Set up the register classes
+  addRegisterClass(MVT::i8, &STM8::GPR8RegClass);
+  addRegisterClass(MVT::i16, &STM8::GPR16RegClass);
+
+  // Compute derived properties from the register classes
+  computeRegisterProperties(STI.getRegisterInfo());
+
+  // TODO: Add possible instructions
+
+  // Provide all sorts of operation actions
+  setStackPointerRegisterToSaveRestore(STM8::R_SP);
+  setBooleanContents(ZeroOrOneBooleanContent);
+  setBooleanVectorContents(ZeroOrOneBooleanContent);
```

Comment:
I do not understand what `setBooleanVectorContents` does, so I can't review this bit for now.

Reply:
Honestly I'm not sure either. I suspect it has something to do with vector operations, but as it is mentioned in AVR and MSP430 I assume that either I'm right and it is required, even if not supported or it is something totally different.